gateway/api/ray.py

```diff
@@ -99,54 +69,118 @@ def create_ray_cluster(
     Returns:
         returns compute resource associated with ray cluster
     """
-    kube_ray_api_server_host = settings.RAY_KUBERAY_API_SERVER_URL
     namespace = settings.RAY_KUBERAY_NAMESPACE
     image = settings.RAY_NODE_IMAGE
-    template_name = settings.RAY_KUBERAY_DEFAULT_TEMPLATE_NAME
-
-    clusters_url = (
-        f"{kube_ray_api_server_host}/apis/v1alpha2/namespaces/{namespace}/clusters"
-    )
-
-    create_compute_template_if_not_exists()
 
     cluster_name = cluster_name or f"{user.username}-{str(uuid.uuid4())[:8]}"
 
-    response = requests.post(
-        clusters_url,
-        json={
-            "name": cluster_name,
-            "namespace": namespace,
-            "user": user.username,
-            "version": "1.9.2",
-            "environment": "DEV",
-            "clusterSpec": {
-                "headGroupSpec": {
-                    "computeTemplate": template_name,
-                    "image": image,
-                    "serviceType": "NodePort",
-                    "rayStartParams": {
-                        "dashboard-host": "0.0.0.0",
-                        "node-ip-address": "$MY_POD_IP",
-                        "port": "6379",
-                    },
-                },
-                "workerGroupSpec": [
-                    {
-                        "groupName": "default-worker-group",
-                        "computeTemplate": template_name,
-                        "image": image,
-                        "replicas": 0,
-                        "minReplicas": 0,
-                        "maxReplicas": 4,
-                        "rayStartParams": {"node-ip-address": "$MY_POD_IP"},
-                    }
-                ],
-            },
-        },
-        timeout=30,
+    config.load_incluster_config()
+    k8s_client = client.api_client.ApiClient()
+    dyn_client = DynamicClient(k8s_client)
+    raycluster_client = dyn_client.resources.get(
+        api_version="v1alpha1", kind="RayCluster"
     )
-    if not response.ok:
+
+    cluster = """
```

Comment:
will do a full review on Monday, but had a question: could we change multiple resources together? i.e. something like:

```yaml
apiVersion: someresource
kind: something
metadata:
  name: one
spec:
  image: imageone:latest
---
apiVersion: someotherresource
kind: something_else
metadata:
  name: two
spec:
  image: imagetwo:latest
```

Reply:
I don't think so.  We have to create the dynamic client for each type of the resource.  We can use the structured client for non-CRD resources.